source/exec.tex

```diff
@@ -5672,3 +5687,424 @@
+such that
+\tcode{r.execute(i, i+1)} for index \tcode{i}
+has effects equivalent to \tcode{f(i, args...)}.
+
+\pnum
+Let \tcode{b} be \tcode{\exposid{BACKEND-OF}(sch)},
+let \tcode{sndr} be the object returned by \tcode{schedule(sch)}, and
+let \tcode{rcvr} be a receiver.
+If \tcode{rcvr} is connected to \tcode{sndr} and
+the resulting operation state is started, then:
+\begin{itemize}
+\item
+If \tcode{sndr} completes successfully,
+then \tcode{b.schedule(r, s)} is called, where
+\begin{itemize}
+\item
+\tcode{r} is a proxy for \tcode{rcvr}
+with base \tcode{system_context_replaceability::receiver_proxy} and
+\item
+\tcode{s} is a preallocated backend storage for \tcode{r}.
+\end{itemize}
+\item
+All other completion operations are forwarded unchanged.
+\end{itemize}
+
+\pnum
+\tcode{parallel_scheduler} provides a customized implementation of
+the \tcode{bulk_chunked} algorithm\iref{exec.bulk}.
+If a receiver \tcode{rcvr} is connected to the sender
+returned by \tcode{bulk_chunked(sndr, pol, shape, f)} and
+the resulting operation state is started, then:
+\begin{itemize}
+\item
+If \tcode{sndr} completes with values \tcode{vals},
+let \tcode{args} be a pack of lvalue subexpressions designating \tcode{vals},
+then \tcode{b.schedule_bulk_chunked(shape, r, s)} is called, where
+\begin{itemize}
+\item
+\tcode{r} is a bulk chunked proxy for \tcode{rcvr}
+with callable \tcode{f} and arguments \tcode{args} and
+\item
+\tcode{s} is a preallocated backend storage for \tcode{r}.
+\end{itemize}
+\item
+All other completion operations are forwarded unchanged.
+\end{itemize}
+\begin{note}
+Customizing the behavior of \tcode{bulk_chunked}
+affects the default implementation of bulk.
+\end{note}
+
+\pnum
+\tcode{parallel_scheduler} provides a customized implementation of
+the bulk_unchunked algorithm\iref{exec.bulk}.
+If a receiver \tcode{rcvr} is connected to the sender
+returned by \tcode{bulk_unchunked(sndr, pol, shape, f)} and
+the resulting operation state is started, then:
+\begin{itemize}
+\item
+If \tcode{sndr} completes with values \tcode{vals},
+let \tcode{args} be a pack of lvalue subexpressions designating \tcode{vals},
+then \tcode{b.schedule_bulk_unchunked(shape, r, s)} is called, where
+\begin{itemize}
+\item
+\tcode{r} is a bulk unchunked proxy for \tcode{rcvr}
+with callable \tcode{f} and arguments \tcode{args} and
+\item
+\tcode{s} is a preallocated backend storage for \tcode{r}.
+\end{itemize}
+\item
+All other completion operations are forwarded unchanged.
+\end{itemize}
+
+\indexlibraryglobal{get_parallel_scheduler}%
+\begin{itemdecl}
+parallel_scheduler get_parallel_scheduler();
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Let \tcode{eb} be the result of \tcode{system_context_replaceability::query_parallel_scheduler_backend()}.
+If \tcode{eb == nullptr} is \tcode{true},
+calls \tcode{terminate}\iref{except.terminate}.
+Otherwise, returns a \tcode{parallel_scheduler} object
```

Comment:
Note that the paper originally said "`terminate()`" and "an `parallel_scheduler`" here (which are wrong and should be fixed, but still, it's all in one commit).